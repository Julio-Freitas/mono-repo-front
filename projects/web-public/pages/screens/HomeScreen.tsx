import { sum } from "@alura/utils/macth";
import { Text } from "@alura/ds";

export function HomeScreen() {
  return (
    <div>
      <Text tag="h1">Home do Projetos</Text>
      <Text tag="p">@alura/utils/math/sum: sum(2,2) = {sum(2, 2)}</Text>
    </div>
  );
}
