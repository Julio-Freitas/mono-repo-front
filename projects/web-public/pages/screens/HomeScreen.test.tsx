import { customRender } from "@alura/test-commons/react-testing-library";

import { HomeScreen } from "./HomeScreen";
import { AllProviders } from "../../components/AllProviders";

// Aqui podemos customizar os valores padrão dos providers para cada teste
const render = customRender((props) => <AllProviders {...props} />);

describe("<HomeScreen />", () => {
  it("renders the page", () => {
    const { container } = render(<HomeScreen />);
    // Para páginas provavemlente não iremos fazer teste de snapshot, mas fica o demo
    expect(container).toMatchSnapshot();
  });
});
