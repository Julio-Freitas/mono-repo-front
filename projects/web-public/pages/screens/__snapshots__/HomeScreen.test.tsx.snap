// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders the page 1`] = `
.c0 {
  color: red;
}

<div>
  Provider Here!
  <div>
    <h1
      class="c0"
    >
      Home do Projetos
    </h1>
    <p
      class="c0"
    >
      @alura/utils/math/sum: sum(2,2) = 
      4
    </p>
  </div>
</div>
`;
